torch/distributed/_sharded_tensor/api.py

```diff
@@ -86,13 +113,14 @@ def __init__(
         self._process_group = (
             process_group
             if process_group is not None
-            else torch.distributed.distributed_c10d._get_default_group()
+            else distributed_c10d._get_default_group()
         )
 
-        if torch.distributed.distributed_c10d._rank_not_in_group(self._process_group):
+        if distributed_c10d._rank_not_in_group(self._process_group):
             raise ValueError(f'Global rank: {dist.get_rank()} not part of process group')
 
         self._local_shards: List[Shard] = []
+        self._remote_shards: List[rpc.RRef] = None
```

Comment:
1) Why is the default value `None` instead of an empty list?
2) `rpc.RRef` -> `rpc.RRef[Shard]`?

Reply:
1. It is mostly for the check in `remote_shards`. If it is None it means RPC was not initialized before ShardedTensor was created and we should throw an error instead of returning an empty list.
2. Will update this.

Reply:
> It is mostly for the check in remote_shards. If it is None it means RPC was not initialized before ShardedTensor was created and we should throw an error instead of returning an empty list.

Thanks for the explanation! Saw the check in L358. I think it's still better to add a comment here to explain the subtle difference here, since readers may ignore the implementation details.

Reply:
Added a separate field to track this instead for clarity